### DISCUSSION TITLE
Flutter: 3GB Cache size with recommended settings

### DISCUSSION THREAD

**Discussion by flodaniel** [codemagic-ci-cd/codemagic-docs#1876] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Try to cache your flutter build with the [config recommended](https://docs.codemagic.io/knowledge-codemagic/caching/) (see below)
2. Observe that the cache is ~3GB, and not used.

```
    cache:
      cache_paths:
        - $HOME/Library/Caches/CocoaPods
        - $HOME/.gradle/caches
        - $FLUTTER_ROOT/.pub-cache
```
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/6661557/3e9a5b44-fd56-4efd-9ef2-ee928137a126)

### Expected results

Cache being a reasonable size that speeds up my builds.

### Actual results

This is the output that i get:
```
Starting to cache dependencies
Skipping /Users/builder/programs/flutter/.pub-cache: path not found.

Creating cache archive...
Created cache archive in 2 min 44 seconds
Skipping cache upload because its size 2.98 GB exceeds maximum allowed size of 1 GB
```

In addition this steps takes very long (~3minutes) and then produces no result that will speed up future builds :(

### Build id (optional)

64aff19b2dd25b9d7f60255f

**Comment by dtrdic** [comment]
Hello @flodaniel, please note that caching in Codemagic is limited to 1GB, as indicated in the warning message

**Reply by flodaniel** [reply]
I understand that. What i do not understand is what I should cache then, as the recommende three paths are not working (one does not exist, the other two are too big). Maybe the docs should be updated in that regard?

**Reply by dtrdic** [reply, marked as answer]
Hello @flodaniel! When it comes to caching in Codemagic, the specific paths you choose to cache depend on your project and its size. While there are recommended paths, it's important to keep in mind that not all projects are the same and the effectiveness of caching can vary. Therefore, caching is not a guaranteed solution to speed up your builds.

For your project, you can consider caching the following paths: 
`$HOME/.pub-cache`
or
`$FLUTTER_ROOT/bin/cache/dart-sdk` These paths are commonly found in Flutter projects.

The caching process takes place during the final step of a successful build, and the cached files are retrieved in subsequent builds. If any of the files within the specified paths change, the cache will be updated accordingly.

To determine the sizes of the files/folders you want to cache, you can connect to a build machine via [VNC](https://docs.codemagic.io/troubleshooting/accessing-builder-machine-via-ssh/) and check their sizes. You can then monitor the amount of cached data through the Codemagic UI.
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/fa3a063f-2bc2-4eba-9357-7c2c7a0cb8c0)

**Reply by flodaniel** [reply]
I will try out `$HOME/.pub-cache` and `$FLUTTER_ROOT/bin/cache/dart-sdk`. I do not understand why I would ever have to cache any node modules in a flutter project though? 

Thanks for your input!

**Reply by dtrdic** [reply]
yes you are correct, mistakenly included it in the response, updated:)
